### PR TITLE
chore: merge blocksuite source code

### PR DESCRIPTION
**Pull request by Saul-Mirone**

### CODE REVIEW

**Comment by Saul-Mirone** [comment]
* **#9213** <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/9213?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/toeverything/AFFiNE/9213?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `canary`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/toeverything?ref=merge-queue-instructions-in-stack-comment&prId=9213).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: oxlint.json [hunk 1]
```diff
@@ -181,6 +186,14 @@
           }
         ]
       }
+    },
+    {
+      "files": ["blocksuite/**/*.ts"],
+      "rules": {
+        "eslint/eqeqeq": "off",
```
Comment (forehalo) [inline oxlint.json:193]: why

File: packages/common/env/src/constant.ts [hunk 2]
```diff
@@ -1,4 +1,6 @@
 // This file should has not side effect
+// oxlint-disable-next-line
+// @ts-ignore FIXME: typecheck error
```
Comment (forehalo) [inline packages/common/env/src/constant.ts:3]: is it easy to fix?

File: oxlint.json (hunk 1 above)
Comment (Saul-Mirone) [reply oxlint.json:193]: a lot of `x != null` in bs source code.

File: packages/common/env/src/constant.ts (hunk 2 above)
Comment (Saul-Mirone) [reply packages/common/env/src/constant.ts:3]: Hard to fix for me, may need some help

**Review by EYHN — APPROVED** [review]

**Comment by Saul-Mirone** [comment]
### Merge activity

* **Dec 20, 2:20 AM EST**: The merge label 'merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=toeverything&repo=AFFiNE) once it meets the requirements.
* **Dec 20, 2:35 AM EST**: The merge label 'merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=toeverything&repo=AFFiNE) once it meets the requirements.
* **Dec 20, 2:35 AM EST**: The merge label 'merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=toeverything&repo=AFFiNE) once it meets the requirements.
* **Dec 20, 2:38 AM EST**: The merge label 'merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=toeverything&repo=AFFiNE) once it meets the requirements.

**Pull request merged**